frontend/packages/dev-console/src/components/pipelines/pipeline-builder/task-sidebar/TaskSidebarParam.tsx

```diff
@@ -13,31 +14,36 @@ const TaskSidebarParam: React.FC<TaskSidebarParamProps> = (props) => {
   const { hasParamError, onChange, resourceParam, taskParam } = props;
   const [dirty, setDirty] = React.useState(false);
 
-  const currentValue = taskParam?.value?.toString() || '';
+  const currentValue = taskParam?.value;
   const emptyIsInvalid = !resourceParam.default;
 
-  const isValid = !(dirty && hasParamError && emptyIsInvalid && currentValue === '');
+  const isValid = !(dirty && hasParamError && emptyIsInvalid && currentValue != null);
+
+  const paramRenderProps: ParameterProps = {
+    currentValue,
+    defaultValue: resourceParam.default,
+    isValid,
+    name: resourceParam.name,
+    onChange,
+    setDirty,
+  };
 
   return (
     <FormGroup
       fieldId={resourceParam.name}
       label={resourceParam.name}
-      helperText={resourceParam.description}
+      helperText={resourceParam.type === 'string' ? resourceParam.description : null}
```

Comment:
Is it by design that there be no helperText for any other param type but string?